### DISCUSSION TITLE
Hide prompts/completions (privacy)

### DISCUSSION THREAD

**Discussion by ziodave** [langfuse/langfuse#1392] [Ideas]
In order to respect the privacy of clients, I think it would be nice to provide a system wide option to prevent the actual content of prompts and completions to be stored in the system.

Pros
Compliant with regulations
Enterprise grade requirement
Optional configuration 

Cons
Harder to diagnose actual prompt / completion when needed, mitigated by the fact that it's an option

**Comment by marcklingen** [comment]
Thanks for raising this, we thought a lot about this already and currently consider the following options:

1. do not send inputs/outputs to langfuse -> this will make tokenization/cost tracking impossible as it should not be done client-side for performance reasons
2. Process but do not store inputs/outputs in Langfuse. Can be used for token counts/costs but are not persisted to the db.
3. Store inputs/outputs but control access via RBAC roles.

Toggle the setting, tbd what makes most sense
- SDK flag
- Project-level setting in Langfuse UI
- Option on each generation logged to Langfuse

**Reply by ziodave** [reply]
Option 2 is feasible for me.
